application/apps/indexer/processor/src/export/mod.rs

```diff
@@ -4,26 +4,34 @@ use futures::StreamExt;
 use indexer_base::config::IndexSection;
 use parsers::{LogMessage, MessageStreamItem};
 use thiserror::Error;
+use tokio_util::sync::CancellationToken;
 
 #[derive(Error, Debug)]
 pub enum ExportError {
     #[error("Configuration error ({0})")]
     Config(String),
     #[error("IO error: {0:?}")]
     Io(#[from] std::io::Error),
+    #[error("Cancelled")]
+    Cancelled,
 }
 
+/// Exporting data as raw into given destination. Would be exported only data
+/// defined in selections as indexes of rows (messages).
+/// Returns (usize, usize) - (exported_count, read_count)
+/// read_count is equal to total count of messages (rows) in source.
 pub async fn export_raw<S, T>(
     mut s: S,
     destination_path: &Path,
-    sections: Vec<IndexSection>,
-) -> Result<usize, ExportError>
+    sections: &Vec<IndexSection>,
+    cancel: &CancellationToken,
+) -> Result<(usize, usize), ExportError>
```

Comment:
returning the read_count seems redundant...alreay part of the argument (sections)

Reply:
actually, we don't need information how much stuff was exported, but we need information about how many rows/messages were read. I've changed it to return only it - count of read messages